### PR TITLE
some optimizations for cert.go

### PR DESCRIPTION
**Pull request by ls0f**

### CODE REVIEW

**Comment by ishitasequeira** [comment]
@ls0f Can we add some description to the PR for the need of the change?

File: util/cert/cert.go [hunk 1]
```diff
@@ -100,11 +99,10 @@ func GetTLSCertificateDataPath() string {
 // filesystem. If ARGOCD_SSH_DATA_PATH environment is set, path is taken from
 // there, otherwise the default will be returned.
 func GetSSHKnownHostsDataPath() string {
-	envPath := os.Getenv(common.EnvVarSSHDataPath)
-	if envPath != "" {
-		return envPath + "/" + common.DefaultSSHKnownHostsName
```
Comment (ls0f) [inline util/cert/cert.go:105]: use filepath.Join to get the final dst path

File: util/cert/cert.go [hunk 2]
```diff
@@ -332,7 +327,7 @@ func GetCertificateForConnect(serverName string) ([]string, error) {
 // mount. This function makes sure that the path returned actually contain
 // at least one valid certificate, and no invalid data.
 func GetCertBundlePathForRepository(serverName string) (string, error) {
```
Comment (ls0f) [inline util/cert/cert.go:329]: use filepath.Join to get the final dst path too.

File: util/cert/cert.go [hunk 3]
```diff
@@ -302,9 +300,6 @@ func ServerNameWithoutPort(serverName string) string {
 // consider it an error and just return empty data.
 func GetCertificateForConnect(serverName string) ([]string, error) {
 	dataPath := GetTLSCertificateDataPath()
-	if !strings.HasSuffix(dataPath, "/") {
```
Comment (ls0f) [inline util/cert/cert.go:305]: since use file path to combine dir, it's unnecessary

**Review by ishitasequeira — APPROVED** [review]
LGTM!

File: util/cert/cert.go [hunk 4]
```diff
@@ -88,8 +88,7 @@ func IsValidHostname(hostname string, fqdn bool) bool {
 // filesystem. If ARGOCD_TLS_DATA_PATH environment is set, path is taken from
 // there, otherwise the default will be returned.
 func GetTLSCertificateDataPath() string {
-	envPath := os.Getenv(common.EnvVarTLSDataPath)
-	if envPath != "" {
```
Comment (ls0f) [inline util/cert/cert.go:92]: more clean code i think

**Review by ishitasequeira — APPROVED** [review]

**Review by crenshaw-dev — APPROVED** [review]
Thanks!

**Pull request merged**